kuma/users/tests/test_templates.py

```diff
@@ -560,66 +608,716 @@ def test_banned_user_one_click_page_template_ban_button(self):
         eq_(len(ban_button), 0)
         eq_(len(ban_other_reasons), 0)
 
+        # For self.testuser2 (banned, no revisions needing to be reverted)
+        # there should be no button on the form and no link to
+        # ban for other reasons
+        # Create some revisions made by self.testuser2 and add a Spam submission for each
+        num_revisions = 3
+        created_revisions = self.create_revisions(
+            num=num_revisions,
+            document=self.document,
+            creator=self.testuser2)
+        for revision in created_revisions:
+            revision.akismet_submissions.add(RevisionAkismetSubmission(sender=self.testuser2, type="spam"))
+
+        ban_url = reverse('users.ban_user_and_cleanup',
+                          kwargs={'user_id': self.testuser2.id})
+        resp = self.client.get(ban_url, follow=True)
+        eq_(200, resp.status_code)
+        page = pq(resp.content)
+
+        revisions_found = page.find('.dashboard-row')
+        ban_button = page.find('#ban-and-cleanup-form button[type=submit]')
+        ban_other_reasons = page.find('#ban-for-other-reasons')
+
+        eq_(len(revisions_found), 3)
+        eq_(len(ban_button), 0)
+        eq_(len(ban_other_reasons), 0)
+
 
 @pytest.mark.bans
 class BanUserAndCleanupSummaryTestCase(SampleRevisionsMixin, UserTestCase):
```

Comment:
I'm assuming all the ``TODO:`` in this section is collaboration on Phase 3 tests.  Normally, I'd want the speculative future code removed, so the PR only covers the changes, but I think this is a special case, so keep them to be uncommented in the next PR.